spec/models/concerns/diagnosis_creation_spec.rb

```diff
@@ -124,4 +124,54 @@
       end
     end
   end
+
+  describe 'prepare_matches_from_solicitation' do
+    let(:diagnosis) { create :diagnosis, solicitation: solicitation }
+    let(:solicitation) { create :solicitation }
+    let(:need) { create :need, diagnosis: diagnosis }
+
+    let!(:expert_subject) do
+      create :expert_subject,
+             institution_subject: create(:institution_subject, institution: institution, subject: the_subject),
+             expert: create(:expert, communes: communes)
+    end
+    let(:institution) { create :institution }
+
+    before do
+      allow(solicitation).to receive(:preselected_institutions).and_return(preselected_institutions)
```

Comment:
J'ai pas compris pourquoi on ne peut pas écrire `solicitation. preselected_institutions`

Reply:
il y a la même plus haut avec `preselected_subjects`

Reply:
(Je suis pas sûr de comprendre la question) On pourrait configurer la solicitation avec des options qui référencent les institutions qu’on veut, mais ça me semblait plus court de faire comme ça que de créer tout les objets. Tu veux que j’essaye?

Reply:
c'est vrai que c'était pas très clair, en fait à quoi ça sert de faire `allow(solicitation).to receive(:preselected_institutions).and_return(preselected_institutions)` au lieu de `solicitation.preselected_institutions`. La c'est plutôt de la compréhension de Rspec ma question 😄

Reply:
`receive(:preselected_institutions)`, c’est le message que va recevoir ma `solicitation` (enfin, la méthode qui sera appelée par la fonction testée), tandis que `and_return(preselected_institutions)` c’est la valeur que je veux qu’elle retourne à la place d’appeler la vraie méthode. Dans le test, j’aurais pu l’appeler `let(:toto)` et écrire `and_return(toto)`.

Reply:
ah oui d'accord, j'ai compris